Pull request to fix Linux Support

jimmon89 opened the pull request:
if you would like, the messages to log on game start are easily removed.
and if you would rather not include instructions to set up the wine prefix, that too is easy to remove, it's on its own push
also, sorry to ninja edit the link to the latest release, the link i put always points to the latest and only the latest release

jimmon89 committed:
Add instructions for creating wine prefix

full command line instructions for creating the wine prefix, installing the dependencies inside the prefix, downloading this launcher, and creating scripts and optionally shortcuts to run it all.

jimmon89 committed:
Several Changes For Linux Support

Fix support for linux under wine without breaking windows support
Fix support for non-M1(intel) Mac OSX under wine
Change launch method depending on system type using `sys.platform.startswith`

The following can be removed if requested.
If OS is Linux and Mac OSX, A warning is added to the log about needing prior setup of wine and related environment variables
If OS is Linux, A warning is added to the log that Linux support is in beta
If OS is Mac OSX, A warning is added to the log that Mac OSX is unsupported
If OS is not Windows, Linux or Mac OSX, an error is logged and program exits

chtheiss commented on duskhaven_launcher.py:
```diff
@@ -564,9 +564,25 @@ def complete_launcher_update(self, new_version_path):
+            logger.info("Proper prior setup of wine and related environment variables is highly recommended")
+            subprocess.Popen(["wine", pathlib.Path(self.configuration["installation_path"]) / "wow.exe"])
+        else:
+            logger.info("ERROR: " + sys.platform + " is completely unsupported!")
```
you can just call logger.error here: `logger.error(f"{sys.platform} is completely unsupported!")`

jimmon89 committed:
Change method when unsupported

change error method `logger.info(` into `logger.error(f` for totally unsupported systems (Meaning not Windows, Linux, or Mac OSX)

jimmon89 commented:
Is that better?
I'm open to any and all alternatives XD

jimmon89 committed:
revert 2nd error log to normal

revert `logger.error(f"Exiting!")` to logger.info("Exiting!")
